### PR TITLE
Registration2

### PR DESCRIPTION
**Pull request by IstrajI**

### CODE REVIEW

File: app/src/main/java/pro/itshark/moneysplitter/di/modules/ActionsModule.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package pro.itshark.moneysplitter.di.modules
+
+import dagger.Module
+import dagger.Provides
+import pro.itshark.moneysplitter.presentation.regLogin.RegLoginActivityViewModel
+import pro.itshark.moneysplitter.presentation.regLogin.login.LoginActions
+import javax.inject.Singleton
+
+@Module
+class ActionsModule {
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/di/modules/ActionsModule.kt]: All

File: app/src/main/java/pro/itshark/moneysplitter/di/modules/ViewModelModule.kt [hunk 2]
```diff
@@ -32,4 +36,22 @@ abstract class ViewModelModule {
...
+    @IntoMap
+    @Singleton
+    @ViewModelKey(RegLoginActivityViewModel::class)
+    abstract fun bindRegLoginViewModel(regLoginActivityViewModel: RegLoginActivityViewModel): ViewModel
+
+    @Binds
+    @IntoMap
+    @Singleton
+    @ViewModelKey(LoginViewModel::class)
+    abstract fun bindLoginActivityViewModel(loginViewModel: LoginViewModel): ViewModel
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/di/modules/ViewModelModule.kt]: Fixed

File: app/src/main/java/pro/itshark/moneysplitter/di/modules/ViewModelModule.kt [hunk 3]
```diff
@@ -32,4 +36,22 @@ abstract class ViewModelModule {
     @IntoMap
     @ViewModelKey(EventsViewModel::class)
     abstract fun bindEventsViewModel(eventsViewModel: EventsViewModel): ViewModel
+
+    @Binds
+    @IntoMap
+    @Singleton
+    @ViewModelKey(RegLoginActivityViewModel::class)
+    abstract fun bindRegLoginViewModel(regLoginActivityViewModel: RegLoginActivityViewModel): ViewModel
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/di/modules/ViewModelModule.kt]: Fixed

File: app/src/main/java/pro/itshark/moneysplitter/di/modules/ViewModelModule.kt [hunk 4]
```diff
@@ -32,4 +36,22 @@ abstract class ViewModelModule {
...
+    @IntoMap
+    @Singleton
+    @ViewModelKey(LoginViewModel::class)
+    abstract fun bindLoginActivityViewModel(loginViewModel: LoginViewModel): ViewModel
+
+    @Binds
+    @IntoMap
+    @Singleton
+    @ViewModelKey(RegistrationViewModel::class)
+    abstract fun bindRegistrationActivityViewModel(registrationViewModel: RegistrationViewModel): ViewModel
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/di/modules/ViewModelModule.kt]: Fixed

File: app/src/main/java/pro/itshark/moneysplitter/di/modules/ActionsModule.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (BlaizMiko) [reply app/src/main/java/pro/itshark/moneysplitter/di/modules/ActionsModule.kt]: Mb will be better split it?

File: app/src/main/java/pro/itshark/moneysplitter/domain/user/UserUseCases.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import io.reactivex.Observable
+import io.reactivex.Single
+import pro.itshark.moneysplitter.model.pojo.UserEntry
+import pro.itshark.moneysplitter.presentation.userProfile.UserModel
+
+interface UserUseCases {
+    fun getUserInfo(): Single<UserModel>
+    fun register(userEntry: UserEntry) : Observable<UserEntry>
+    fun login(userEntry: UserEntry) : Observable<UserEntry>
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/domain/user/UserUseCases.kt]: Fixed

File: app/src/main/java/pro/itshark/moneysplitter/domain/user/UserUseCases.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package pro.itshark.moneysplitter.domain.user
+
+import io.reactivex.Observable
+import io.reactivex.Single
+import pro.itshark.moneysplitter.model.pojo.UserEntry
+import pro.itshark.moneysplitter.presentation.userProfile.UserModel
+
+interface UserUseCases {
+    fun getUserInfo(): Single<UserModel>
+    fun register(userEntry: UserEntry) : Observable<UserEntry>
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/domain/user/UserUseCases.kt]: Fixed

File: app/src/main/java/pro/itshark/moneysplitter/presentation/regLogin/login/LoginFragment.kt [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+class LoginFragment: Fragment() {
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+    lateinit var viewModel: LoginViewModel
+    lateinit var loginActions: LoginActions
+
+    private val stateObserver = Observer<LoginScreenState> { state ->
+        state?.let {
+            when (state) {
+                is LoginErrorState -> text_view_login_error_message.visibility = View.VISIBLE
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/presentation/regLogin/login/LoginFragment.kt]: Fixed

File: app/src/main/java/pro/itshark/moneysplitter/presentation/regLogin/registration/RegistrationFragment.kt [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+class RegistrationFragment: Fragment() {
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+    lateinit var viewModel: RegistrationViewModel
+
+    private val stateObserver = Observer<RegistrationScreenStates> { state ->
+        state?.let {
+            when (state) {
+                is RegistrationSuccessState -> text_view_registration_error_message.visibility = View.INVISIBLE
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/presentation/regLogin/registration/RegistrationFragment.kt]: Fixed

**Pull request merged**